### PR TITLE
NT-376 - Android physical back button compability

### PR DESCRIPTION
**Pull request by eiriko2303**
Back button can do the following:
- Go back from pages
- Close side menu
- Close dropdown
- Close the customkeyboard

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -47,97 +53,150 @@ export class MyApp {
...
+      } else {
+        this.checkWhichPageToPop();
+      }
+    })
+  }
+
+  checkWhichPageToPop() {
+    let max = pages.length;
+    for (let i = 0; i < max; i++) {
+      let n = this.navCtrl;
```
Comment (noro5) [inline src/app/app.component.ts]: Maybe that definition should be outside the For loop?

File: src/components/week-picker-dropdown/week-picker-dropdown.component.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<div class="dropdown-list" *ngIf="showDropdown">
+<div class="dropdown-list" *ngIf="showDropDown">
```
[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/components/week-picker-dropdown/week-picker-dropdown.component.html]: I have changed the naming to camelcase cause i thought it should be like that, but I can change it back? :)

File: src/app/app.component.ts (hunk 1 above)
Comment (eiriko2303) [reply src/app/app.component.ts]: Agree, fixing it now :)

File: src/app/app.component.ts [hunk 3]
```diff
@@ -47,97 +53,150 @@ export class MyApp {
...
+        return;
+      } else if(this.showingKeyboard) {
+        this.ngRedux.dispatch(setShowKeyboard(false));
+      } else {
+        this.checkWhichPageToPop();
+      }
+    })
+  }
+
+  checkWhichPageToPop() {
```
Comment (reg0) [inline src/app/app.component.ts]: I would name it differently, "check" suggests it returns boolean, and doesn't do any other action, but that's not the case

how about ```popPageIfPossible()``` ?

File: src/components/week-picker/week-picker.ts [hunk 4]
```diff
@@ -124,15 +127,15 @@ export class WeekPickerComponent {
   }
 
   showDropDown() {
-    if(this.weekPickerDropdown.showDropdown) {
-      this.weekPickerDropdown.showDropdown = false;
+    if(this.isDropDownShown) {
```
Comment (reg0) [inline src/components/week-picker/week-picker.ts]: method should be renamed to toggleDropdown, if it also hides it

File: src/app/app.component.ts (hunk 3 above)
Comment (eiriko2303) [reply src/app/app.component.ts]: Agree :)

File: src/components/week-picker/week-picker.ts (hunk 4 above)
Comment (eiriko2303) [reply src/components/week-picker/week-picker.ts]: 👍

File: src/app/app.module.ts [hunk 5]
```diff
@@ -96,6 +96,8 @@ import { ActionButtonComponent } from '../components/action-button/action-button
 import { ApprovalHistoryPage } from '../pages/approval-history/approval-history';
 import { BrowserSupport } from './browserSupport';
 import { DirectRegistrationModal } from '../components/direct-registration-modal/direct-registration-modal.component';
+import { InitalizationService } from '../services/initialization.service';
```
[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/app/app.module.ts]: Thanks! Fixed and tested it now :)

**Pull request merged**